crates/wallet/src/wallet/mod.rs

```diff
@@ -2014,20 +2026,22 @@ impl Wallet {
             return (must_spend, vec![]);
         }
 
+        let canon_txs = self
+            .indexed_graph
+            .graph()
+            .list_canonical_txs(&self.chain, chain_tip)
+            .map(|canon_tx| (canon_tx.tx_node.txid, canon_tx))
+            .collect::<HashMap<Txid, _>>();
+
         let satisfies_confirmed = may_spend
             .iter()
             .map(|u| -> bool {
                 let txid = u.0.outpoint.txid;
-                let tx = match self.indexed_graph.graph().get_tx(txid) {
-                    Some(tx) => tx,
-                    None => return false,
-                };
-                let chain_position = match self.indexed_graph.graph().get_chain_position(
-                    &self.chain,
-                    chain_tip,
-                    txid,
-                ) {
-                    Some(chain_position) => chain_position.cloned(),
+                let (chain_position, tx) = match canon_txs.get(&txid) {
+                    Some(CanonicalTx {
+                        chain_position,
+                        tx_node,
+                    }) => (chain_position, tx_node.tx.clone()),
```

Comment:
In 5456f8b4 feat(chain)!: rm `get_chain_position` and associated methods

```suggestion
                    }) => (chain_position, &tx_node.tx),
```

Reply:
This line no longer exists.